account/signer.go

```diff
@@ -13,13 +14,23 @@ type Signer struct {
 	publicKey string
 }
 
-func NewSigner(privateKey *big.Int) (*Signer, error) {
+func NewSigner(privateKeyHex string, existingKeystore *MemKeystore) (*Signer, error) {
+	privateKey, ok := new(big.Int).SetString(privateKeyHex, 16)
+	if !ok {
+		return nil, fmt.Errorf("invalid private key format")
+	}
+
 	pubKey, err := getPublicKey(privateKey)
 	if err != nil {
 		return nil, err
 	}
 
-	keyStore := SetNewMemKeystore(pubKey, privateKey)
+	var keyStore *MemKeystore
```

Comment:
I believe you confused something here, the existing MemKeyStore was for the Put method. The Put method should accept a privKey and a MemKeyStore pointer as parameter. Let's remove this code from this method

Reply:
done